File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int max_a = 0;
        int l=0, r=height.size()-1;
        while(r>=l){
            int area = (r-l)*min(height[l],height[r]);
            max_a = max(max_a, area);
            if(height[l]>height[r]){
                r--;
            }else{ l++; }
        }
        return max_a;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int n = height.length;
        int right = n-1;
        int left = 0;
        int max_Area = 0;
        while(right>=left){
            int area = (right-left)*Math.min(height[right], height[left]);
            max_Area=Math.max(max_Area, area);
            if(height[left]>height[right]){ right--;}
            else{ left++;}
        }
        return max_Area;
    }
}